staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/types.go

```diff
@@ -66,6 +66,21 @@ type ListOptions struct {
 	// it does not recognize and will return a 410 error if the token can no longer be used because
 	// it has expired.
 	Continue string
+
+	// SendInitialEvents when set together with Watch option, begins the watch stream with synthetic "Added" events of all resources.
+	// Followed by a synthetic "Bookmark" event containing a ResourceVersion after which the server continues streaming events.
+	// This option is meant to be used with the resourceVersion parameter with the following semantics:
+	//
+	// - sendInitialEvent and RV>0 - provides the initial events and when we reach the desired RV,
+	//   the synthetic bookmark event. Note that the desired RV must be >= the current revision the server knows about.
+	//   If the desired RV is older than the current revision the server is aware of, the TooLargeResourceVersionError is returned.
```

Comment:
That's not true - if the desired RV is older, then we simply return the more fresh state (which is fine according to "not older then" semantic. 
The TooLargeResourceVersionError is returned if the desired RV is actually more recent than anything server is aware of.

Reply:
I just copied your previous comment https://github.com/kubernetes/kubernetes/pull/110960#discussion_r930792587

I have just checked and I think that we actually miss `TooLargeResourceVersionError` case. I think we should change the `getAllEventsSinceLocked` method with:
```
if resourceVersion > w.resourceVersion {
		return nil, storage.NewTooLargeResourceVersionError(resourceVersion, w.resourceVersion, resourceVersionTooHighRetrySeconds)
	}
```